### PR TITLE
*: add delete file in range

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay 

Must use rocksdb master now.

### CODE REVIEW

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 //
 
 extern crate libc;
+extern crate tempdir;
```
Comment (BusyJay) [inline librocksdb_sys/src/lib.rs]: May need a `#[cfg(test)]`

File: src/rocksdb.rs [hunk 2]
```diff
@@ -812,6 +812,50 @@ impl DB {
...
         sizes
     }
+
+    pub fn delete_file_in_range(&self,
+                                start_key: &[u8],
+                                limit_key: &[u8])
+                                -> Result<(), String> {
+        unsafe {
+            let mut err: *const i8 = 0 as *const i8;
+            let err_ptr: *mut *const i8 = &mut err;
```
Comment (BusyJay) [inline src/rocksdb.rs]: `err_ptr` seems redundant.

File: src/rocksdb.rs [hunk 3]
```diff
@@ -812,6 +812,50 @@ impl DB {
         }
         sizes
     }
+
+    pub fn delete_file_in_range(&self,
+                                start_key: &[u8],
+                                limit_key: &[u8])
```
Comment (BusyJay) [inline src/rocksdb.rs]: `end_key` or `limit_key`?

File: src/rocksdb.rs [hunk 4]
```diff
@@ -1143,4 +1187,4 @@ fn snapshot_test() {
     }
     let opts = Options::new();
     assert!(DB::destroy(&opts, path).is_ok());
-}
+}
```
Comment (BusyJay) [inline src/rocksdb.rs]: leave a blank line.

**Comment by BusyJay** [comment]
rest LGTM

File: src/rocksdb.rs (hunk 3 above)
Comment (siddontang) [inline src/rocksdb.rs]: rocksdb uses limit_key for C API>

File: src/rocksdb.rs (hunk 2 above)
Comment (siddontang) [inline src/rocksdb.rs]: Em, just refer other functions. You can see get, put, merge all use this way. Let's keep the same.

File: librocksdb_sys/src/lib.rs [hunk 5]
```diff
@@ -147,7 +149,8 @@ extern "C" {
     pub fn rocksdb_options_set_bytes_per_sync(options: DBOptions, bytes: u64);
     pub fn rocksdb_options_set_disable_data_sync(options: DBOptions,
                                                  v: c_int);
-    pub fn rocksdb_options_set_allow_os_buffer(options: DBOptions, is_allow: bool);
+    pub fn rocksdb_options_set_allow_os_buffer(options: DBOptions,
```
Comment (ngaut) [inline librocksdb_sys/src/lib.rs]: Does this api related to the PR?

File: librocksdb_sys/src/lib.rs [hunk 6]
```diff
@@ -168,7 +171,8 @@ extern "C" {
                                                      bytes: u64);
     pub fn rocksdb_options_set_target_file_size_multiplier(options: DBOptions,
                                                            mul: c_int);
-    pub fn rocksdb_options_set_max_bytes_for_level_base(options: DBOptions, bytes: u64);
+    pub fn rocksdb_options_set_max_bytes_for_level_base(options: DBOptions,
```
Comment (ngaut) [inline librocksdb_sys/src/lib.rs]: Ditto

File: librocksdb_sys/src/lib.rs (hunk 5 above)
Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: rustfmt formats it.

File: librocksdb_sys/src/lib.rs (hunk 6 above)
Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: rustfmt formats it.

It is so tiny so I think can be in this PR.

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

File: src/rocksdb.rs [hunk 7]
```diff
@@ -831,6 +831,46 @@ impl DB {
...
+        unsafe {
+            let mut err: *const i8 = 0 as *const i8;
+            let err_ptr: *mut *const i8 = &mut err;
+
+            rocksdb_ffi::rocksdb_delete_file_in_range(self.inner,
+                                        r.start_key.as_ptr(),
+                                        r.start_key.len() as size_t,
+                                        r.end_key.as_ptr(),
+                                        r.end_key.len() as size_t,
+                                        err_ptr);
```
Comment (BusyJay) [inline src/rocksdb.rs]: Use `&mut err` instead.

File: src/rocksdb.rs [hunk 8]
```diff
@@ -831,6 +831,46 @@ impl DB {
         sizes
     }
 
+    pub fn delete_file_in_range(&self, r: Range) -> Result<(), String> {
```
Comment (BusyJay) [inline src/rocksdb.rs]: I prefer pass the actual start key and end key explicitly. Create a new range seems redundant.

**Comment by siddontang** [comment]
PTAL

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**